packages/mobile/src/account/Profile.tsx

```diff
@@ -1,64 +1,111 @@
 import fontStyles from '@celo/react-components/styles/fonts'
-import * as React from 'react'
-import { WithTranslation } from 'react-i18next'
-import { ScrollView, StyleSheet, Text, View } from 'react-native'
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useLayoutEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { ScrollView, StyleSheet, View } from 'react-native'
+import * as RNFS from 'react-native-fs'
 import { SafeAreaView } from 'react-native-safe-area-context'
-import { connect } from 'react-redux'
-import { setName } from 'src/account/actions'
-import { UserContactDetails } from 'src/account/reducer'
-import { userContactDetailsSelector } from 'src/account/selectors'
-import { Namespaces, withTranslation } from 'src/i18n'
-import { RootState } from 'src/redux/reducers'
+import { useDispatch, useSelector } from 'react-redux'
+import { saveNameAndPicture } from 'src/account/actions'
+import { nameSelector, pictureSelector } from 'src/account/selectors'
+import { showError, showMessage } from 'src/alert/actions'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import CancelButton from 'src/components/CancelButton'
+import i18n, { Namespaces } from 'src/i18n'
+import { emptyHeader } from 'src/navigator/Headers'
+import { Screens } from 'src/navigator/Screens'
+import { TopBarTextButton } from 'src/navigator/TopBarButton'
+import { StackParamList } from 'src/navigator/types'
+import PictureInput from 'src/onboarding/registration/PictureInput'
+import { saveProfilePicture } from 'src/utils/image'
+import Logger from 'src/utils/Logger'
 
-interface StateProps {
-  name: string | null
-  userContact: UserContactDetails
-}
+type Props = StackScreenProps<StackParamList, Screens.Profile>
 
-interface DispatchProps {
-  setName: typeof setName
-}
+function Profile({ navigation, route }: Props) {
+  const { t } = useTranslation(Namespaces.accountScreen10)
+  const [newName, setNewName] = useState(useSelector(nameSelector) ?? '')
+  const picturePath = useSelector(pictureSelector)
+  const [newPictureUri, setNewPictureUri] = useState(picturePath || null)
 
-// tslint:disable-next-line: no-empty-interface
-interface OwnProps {}
+  const dispatch = useDispatch()
 
-type Props = OwnProps & StateProps & DispatchProps & WithTranslation
-const mapStateToProps = (state: RootState) => {
-  return {
-    name: state.account.name,
-    userContact: userContactDetailsSelector(state),
+  useLayoutEffect(() => {
+    const onSave = () => {
+      dispatch(saveNameAndPicture(newName, newPictureUri))
+      // TODO: Save name and picture on CIP-8.
+      dispatch(showMessage(t('namePictureSaved')))
+      navigation.goBack()
+
+      // Delete old profile pictures if necessary.
+      if (picturePath && picturePath !== newPictureUri) {
+        RNFS.unlink(picturePath).catch((e) => {
+          Logger.error('Error deleting old profile picture:', e)
+        })
+      }
+    }
```

Comment:
nit: it feels this logic should go in a saga

Reply:
Having a saga for this would add boilerplate and indirection in the logic, which would make it a bit harder to read IMO since we'd have to jump through several files to find this and since the logic is quite simple I feel like just leaving it here makes sense.

Reply:
Yes it's debatable, I tend to prefer grouping more complicated business logic in saga / helper rather than components.
I saw CIP-8 action will be needed here so it seems this could be a good candidate. But we'll see when we actually do it.